--- origsrc/eog-2.24.2/data/eog.pc.in	2008-10-20 11:31:55.000000000 -0500
+++ src/eog-2.24.2/data/eog.pc.in	2008-12-14 14:39:21.931375000 -0600
@@ -9,3 +9,4 @@
 Requires: gtk+-2.0, libgnomeui-2.0 libglade-2.0, gnome-vfs-2.0, gconf-2.0
 Version: @VERSION@
 Cflags: -I${includedir}
+Libs: -L${libdir} -leog
--- origsrc/eog-2.24.3.1/doc/reference/Makefile.am	2008-12-25 15:24:11.000000000 -0600
+++ src/eog-2.24.3.1/doc/reference/Makefile.am	2009-02-03 00:15:39.238125000 -0600
@@ -90,10 +90,6 @@ BINDING_LIBS = $(top_builddir)/bindings/
 MORE_GTKDOC_LIBS = $(PYTHON_LIBS)
 endif
 
-if ENABLE_JPEG
-jpeg_LIB = $(top_builddir)/jpegutils/libeog-jpegutils.a
-endif
-
 GTKDOC_LIBS=\
 	$(top_builddir)/src/libeog.la 	\
 	$(top_builddir)/cut-n-paste/toolbar-editor/libtoolbareditor.la \
--- origsrc/eog-2.24.2/jpegutils/Makefile.am	2008-10-20 11:31:55.000000000 -0500
+++ src/eog-2.24.2/jpegutils/Makefile.am	2008-12-14 14:39:21.947000000 -0600
@@ -1,9 +1,9 @@
 INCLUDES = $(EOG_CFLAGS) \
 	$(EXIF_CFLAGS)
 
-noinst_LIBRARIES = libeog-jpegutils.a
+noinst_LTLIBRARIES = libeog-jpegutils.la
 
-libeog_jpegutils_a_SOURCES =		\
+libeog_jpegutils_la_SOURCES =		\
 	transupp.c			\
 	transupp.h
 
--- origsrc/eog-2.24.3/src/Makefile.am	2008-10-20 11:31:54.000000000 -0500
+++ src/eog-2.24.3/src/Makefile.am	2009-02-02 14:39:21.993875000 -0600
@@ -1,15 +1,12 @@
 if ENABLE_JPEG
-jpeg_LIB = $(top_builddir)/jpegutils/libeog-jpegutils.a
-jpeg_DEPS = $(top_builddir)/jpegutils/libeog-jpegutils.a
+jpeg_LIB = $(top_builddir)/jpegutils/libeog-jpegutils.la
 endif
 
-toolbar_LIB = $(top_builddir)/cut-n-paste/toolbar-editor/libtoolbareditor.la
-toolbar_DEPS = $(top_builddir)/cut-n-paste/toolbar-editor/libtoolbareditor.la
+toolbar_LIB = $(top_builddir)/cut-n-paste/toolbar-editor/libtoolbareditor.la
 
-screensaver_LIB = $(top_builddir)/cut-n-paste/totem-screensaver/libtotemscrsaver.la
-screensaver_DEPS = $(top_builddir)/cut-n-paste/totem-screensaver/libtotemscrsaver.la
+screensaver_LIB = $(top_builddir)/cut-n-paste/totem-screensaver/libtotemscrsaver.la
 
-noinst_LTLIBRARIES = libeog.la
+lib_LTLIBRARIES = libeog.la
 
 bin_PROGRAMS = eog
 
@@ -135,11 +132,13 @@
 endif
 
 if HAVE_EXEMPI
+if !HAVE_EXIF
 INST_H_FILES +=				\
 	eog-exif-details.h
 libeog_la_SOURCES +=			\
 	eog-exif-details.c
 endif
+endif
 
 libeog_la_CFLAGS =						\
 	-I$(top_srcdir)/jpegutils				\
@@ -155,6 +154,10 @@
 	-DEOG_PLUGIN_DIR=\""$(libdir)/eog/plugins"\"
 
 libeog_la_LIBADD = \
+	$(LIBJPEG)			\
+	$(toolbar_LIB)			\
+	$(screensaver_LIB)		\
+	$(jpeg_LIB)			\
 	$(EOG_LIBS) \
 	-lz
 
@@ -173,12 +176,6 @@
 
 eog_SOURCES = main.c
 
-eog_DEPENDENCIES = 	    \
-	libeog.la	    \
-	$(jpeg_DEPS)	    \
-	$(toolbar_DEPS)     \
-	$(screensaver_DEPS)
-
 eog_CFLAGS = 							\
 	-I$(top_srcdir)/cut-n-paste/toolbar-editor		\
 	-I$(top_srcdir)/cut-n-paste/totem-screensaver		\
@@ -189,11 +186,7 @@
 eog_LDADD = 				\
 	libeog.la			\
 	$(EOG_LIBS)			\
-	$(X_LIBS)			\
-	$(LIBJPEG)			\
-	$(toolbar_LIB)			\
-	$(screensaver_LIB)		\
-	$(jpeg_LIB)
+	$(X_LIBS)
 
 eog_LDFLAGS = -export-dynamic -no-undefined -export-symbols-regex "^[[^_]].*"
 
--- origsrc/eog-2.24.2/plugins/fullscreen/Makefile.am	2008-10-20 11:31:53.000000000 -0500
+++ src/eog-2.24.2/plugins/fullscreen/Makefile.am	2008-12-14 15:39:58.587625000 -0600
@@ -14,9 +14,9 @@
 	eog-fullscreen-plugin.c	
 
 libfullscreen_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-libfullscreen_la_LIBADD = $(EOG_LIBS)
+libfullscreen_la_LIBADD = $(top_builddir)/src/libeog.la $(EOG_LIBS)
 
 # Plugin Info
 
--- origsrc/eog-2.24.2/plugins/reload/Makefile.am	2008-10-20 11:31:54.000000000 -0500
+++ src/eog-2.24.2/plugins/reload/Makefile.am	2008-12-14 15:40:14.447000000 -0600
@@ -14,9 +14,9 @@
 	eog-reload-plugin.c	
 
 libreload_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-libreload_la_LIBADD = $(EOG_LIBS)
+libreload_la_LIBADD = $(top_builddir)/src/libeog.la $(EOG_LIBS)
 
 # Plugin Info
 
--- origsrc/eog-2.24.2/plugins/statusbar-date/Makefile.am	2008-10-20 11:31:54.000000000 -0500
+++ src/eog-2.24.2/plugins/statusbar-date/Makefile.am	2008-12-14 15:40:22.634500000 -0600
@@ -13,9 +13,9 @@
 	eog-statusbar-date-plugin.c	
 
 libstatusbar_date_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-libstatusbar_date_la_LIBADD = $(EOG_LIBS)
+libstatusbar_date_la_LIBADD = $(top_builddir)/src/libeog.la $(EOG_LIBS)
 
 # Plugin Info
 
