--- origsrc/eog-2.26.0/data/eog.pc.in	2009-02-23 13:57:30.000000000 -0600
+++ src/eog-2.26.0/data/eog.pc.in	2009-04-02 23:04:58.478201700 -0500
@@ -9,3 +9,4 @@ Description: The GNOME Image Viewer
 Requires: gtk+-2.0, gconf-2.0
 Version: @VERSION@
 Cflags: -I${includedir}
+Libs: -L${libdir} -leog
--- origsrc/eog-2.26.0/doc/reference/Makefile.am	2009-02-04 15:48:59.000000000 -0600
+++ src/eog-2.26.0/doc/reference/Makefile.am	2009-04-02 23:04:29.337576700 -0500
@@ -93,19 +93,12 @@ INCLUDES= \
 	-I$(top_srcdir)/cut-n-paste/totem-screensaver		\
 	$(EOG_CFLAGS)
 
-if ENABLE_JPEG
-jpeg_LIB = $(top_builddir)/jpegutils/libeog-jpegutils.la
-endif
-
 GTKDOC_LIBS=\
 	$(top_builddir)/src/libeog.la 	\
-	$(top_builddir)/cut-n-paste/toolbar-editor/libtoolbareditor.la \
-	$(top_builddir)/cut-n-paste/totem-screensaver/libtotemscrsaver.la \
 	$(BINDING_LIBS)			\
 	$(LIBJPEG)			\
 	$(X_LIBS)			\
 	$(EOG_LIBS)			\
-	$(jpeg_LIB)			\
 	$(MORE_GTKDOC_LIBS)
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
--- origsrc/eog-2.26.0/plugins/fullscreen/Makefile.am	2009-02-23 13:57:29.000000000 -0600
+++ src/eog-2.26.0/plugins/fullscreen/Makefile.am	2009-04-02 23:03:15.462576700 -0500
@@ -14,9 +14,9 @@ libfullscreen_la_SOURCES = \
 	eog-fullscreen-plugin.c
 
 libfullscreen_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-libfullscreen_la_LIBADD = $(EOG_LIBS)
+libfullscreen_la_LIBADD = $(top_builddir)/src/libeog.la $(EOG_LIBS)
 
 # Plugin Info
 
--- origsrc/eog-2.26.0/plugins/reload/Makefile.am	2009-02-23 13:57:29.000000000 -0600
+++ src/eog-2.26.0/plugins/reload/Makefile.am	2009-04-02 23:03:34.212576700 -0500
@@ -14,9 +14,9 @@ libreload_la_SOURCES = \
 	eog-reload-plugin.c
 
 libreload_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-libreload_la_LIBADD = $(EOG_LIBS)
+libreload_la_LIBADD = $(top_builddir)/src/libeog.la $(EOG_LIBS)
 
 # Plugin Info
 
--- origsrc/eog-2.26.0/plugins/statusbar-date/Makefile.am	2009-02-23 13:57:29.000000000 -0600
+++ src/eog-2.26.0/plugins/statusbar-date/Makefile.am	2009-04-02 23:03:43.821951700 -0500
@@ -13,9 +13,9 @@ libstatusbar_date_la_SOURCES = \
 	eog-statusbar-date-plugin.c
 
 libstatusbar_date_la_LDFLAGS = \
-	-avoid-version -module
+	-avoid-version -module -no-undefined
 
-libstatusbar_date_la_LIBADD = $(EOG_LIBS)
+libstatusbar_date_la_LIBADD = $(top_builddir)/src/libeog.la $(EOG_LIBS)
 
 # Plugin Info
 
--- origsrc/eog-2.26.0/src/Makefile.am	2009-02-23 13:57:29.000000000 -0600
+++ src/eog-2.26.0/src/Makefile.am	2009-04-02 23:02:16.790701700 -0500
@@ -6,7 +6,7 @@ toolbar_LIB = $(top_builddir)/cut-n-past
 
 screensaver_LIB = $(top_builddir)/cut-n-paste/totem-screensaver/libtotemscrsaver.la
 
-noinst_LTLIBRARIES = libeog.la
+lib_LTLIBRARIES = libeog.la
 
 bin_PROGRAMS = eog
 
@@ -133,11 +133,13 @@ libeog_la_SOURCES += 			\
 endif
 
 if HAVE_EXEMPI
+if !HAVE_EXIF
 INST_H_FILES +=				\
 	eog-exif-details.h
 libeog_la_SOURCES +=			\
 	eog-exif-details.c
 endif
+endif
 
 libeog_la_CFLAGS =						\
 	-I$(top_srcdir)/jpegutils				\
@@ -153,6 +155,10 @@ libeog_la_CFLAGS =						\
 	-DEOG_PLUGIN_DIR=\""$(libdir)/eog/plugins"\"
 
 libeog_la_LIBADD = \
+	$(toolbar_LIB)			\
+	$(screensaver_LIB)		\
+	$(jpeg_LIB)			\
+	$(LIBJPEG)			\
 	$(EOG_LIBS)
 
 if ENABLE_PYTHON
@@ -180,11 +186,7 @@ eog_CFLAGS = 							\
 eog_LDADD = 				\
 	libeog.la			\
 	$(EOG_LIBS)			\
-	$(X_LIBS)			\
-	$(LIBJPEG)			\
-	$(toolbar_LIB)			\
-	$(screensaver_LIB)		\
-	$(jpeg_LIB)
+	$(X_LIBS)
 
 eog_LDFLAGS = -export-dynamic -no-undefined -export-symbols-regex "^[[^_]].*"
 
